### PR TITLE
feat: add unresponsive timeout for fred

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Enable `check-unresponsive` in fred and update the version to `7.1.2`

## Motivation and Context

This will retry the command automatically every second(ref: https://github.com/aembke/fred.rs/blob/52bff989263b8bb27030be4e20fd313e65608b44/examples/globals.rs#L22) until `unresponsive_timeout` is exhausted. For retry it tries if one succeeds other will be discarded(ref: https://github.com/aembke/fred.rs/blob/52bff989263b8bb27030be4e20fd313e65608b44/src/router/utils.rs#L566) .

## How did you test it?

- Sanity Test with Payments and Refunds.

**This change cannot be tested since we cannot reproduce the situation in sandbox**

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

**Review by akshay-97 — DISMISSED** [review]

File: config/config.example.toml [hunk 1]
```diff
@@ -55,7 +55,7 @@ stream_read_count = 1      # Default number of entries to read from stream if no
 auto_pipeline = true       # Whether or not the client should automatically pipeline commands across tasks when possible.
 disable_auto_backpressure = false # Whether or not to disable the automatic backpressure features when pipelining is enabled.
 max_in_flight_commands = 5000 # The maximum number of in-flight commands (per connection) before backpressure will be applied.
-default_command_timeout = 0  # An optional timeout to apply to all commands.
+default_command_timeout = 30  # An optional timeout to apply to all commands.
```
Comment (SanchithHegde) [inline config/config.example.toml:58]: Please include the units, i.e., seconds in the comment.

File: crates/redis_interface/src/types.rs [hunk 2]
```diff
@@ -57,6 +57,7 @@ pub struct RedisSettings {
     pub max_in_flight_commands: u64,
     pub default_command_timeout: u64,
     pub max_feed_count: u64,
+    pub unresponsive_timeout: u64,
```
Comment (SanchithHegde) [inline crates/redis_interface/src/types.rs:60]: Include this in `config.example.toml` with a comment, also include the unit.

And add this in the other files like `development.toml`, `docker_compose.toml` and `loadtest/config/development.toml`.

Comment (dracarys18) [reply crates/redis_interface/src/types.rs:60]: Done

**Review by SanchithHegde — DISMISSED** [review]

**Review by akshay-97 — DISMISSED** [review]

**Review by jarnura — DISMISSED** [review]

File: config/config.example.toml
Comment (SanchithHegde) [inline config/config.example.toml:1]: Please make the same changes in `config/deployments/env_specific.toml` file as well, and include comments.

Comment (dracarys18) [reply config/config.example.toml:1]: Done

**Review by SanchithHegde — APPROVED** [review]

**Review by akshay-97 — APPROVED** [review]

**Pull request merged**